src/AutorestSwift/Models/Complex/ParameterType.swift

```diff
@@ -176,7 +184,11 @@ enum ParameterType: Codable {
              .sealedChoice:
             return "\(value).rawValue"
         case .array:
-            return "\(value).map { String($0) }.joined(separator: \"\(delimiter)\") "
+            if explode {
+                return "\(value)"
+            } else {
+                return "\(value).map { String($0) }.joined(separator: \"\(delimiter)\") "
+            }
```

Comment:
I'm confused here. If I had `arrayQuery=["A", "B", "C"]` I would expect the following:

```swift
if explode:
  return "arrayQuery=A&arrayQuery=B&arrayQuery=C"
else
  return "A,B,C"
```

Reply:
Your expectation is correct. The construction of the query parameters is done in the stencil file:

from templates/OperationOptionsSnippet.stencil

```

{% for param in op.params.explodeQuery.optional %}
if let {{ param.key }} = {{ param.path }}{{ param.key }} {
    {{ param.value}}.forEach {
        queryParams.append("{{ param.key }}", $0)
    }
}
{% endfor %}

```